src/docker/functions/container/create-container.ts

```diff
@@ -79,10 +79,14 @@ const getExtraHosts = (extraHosts: ExtraHost[]): string[] => {
   return extraHosts.map((extraHost) => `${extraHost.host}:${extraHost.ipAddress}`);
 };
 
-const getPortBindings = (exposedPorts: Port[]): DockerodePortBindings => {
+const getPortBindings = (exposedPorts: PortWithOptionalBinding[]): DockerodePortBindings => {
   const dockerodePortBindings: DockerodePortBindings = {};
   for (const exposedPort of exposedPorts) {
-    dockerodePortBindings[exposedPort] = [{ HostPort: "0" }];
+    if (typeof exposedPort === "number") {
```

Comment:
in the same way that we have `getContainerPort` which does the type checking, perhaps we could also add `getHostPort`? that way we fully encapsulate the type differences

Reply:
I think that makes sense. Should it just return null if `typeof port === 'number'`?

Reply:
Ah, I see. Added in a27f11b.